### PR TITLE
[Search] Add basic functionality for tree search

### PR DESCRIPTION
**Pull request by nikkisuzani**
## Summary
We created a search functionality feature that allows a user to search a bunch of Cards of trees by either ID or by name. 
## Test Plan
N/A
## Other
Please look at TODO: fix navigate on TreeScreen because we were unsure how to make the navigation work to SearchScreen and it has a weird error. 
### Next Steps
Styling Search Cards and clean up default template stuff :) 
### Online Sources
https://callstack.github.io/react-native-paper/card.html
https://callstack.github.io/react-native-paper/searchbar.html
## Screenshots and Tests
After Screenshots:
IMG_5872
IMG_5873
IMG_5874
CC: @mylesdomingo

### CODE REVIEW

File: src/screens/TreeScreen.tsx
Comment (nikkisuzani) [inline src/screens/TreeScreen.tsx:22]: Navigate is weird??

File: src/navigation/TabNavigator.tsx
Comment (nikkisuzani) [inline src/navigation/TabNavigator.tsx:73]: We created a new tab for Search but we should probably restructure Navigation at some point @mylesdomingo

Comment (mylesdomingo) [reply src/navigation/TabNavigator.tsx:73]: it depends on the designs I need finalized navigation flows @kennethlien

File: App.tsx
Comment (mylesdomingo) [inline App.tsx:18]: I made these changes in a separate PR that is merged, can you pull from master and resolve any conflicts? #17

File: src/components/SearchCard.tsx
Comment (mylesdomingo) [inline src/components/SearchCard.tsx:7]: can you use the `Tree` type defined in .tsx?

File: src/database/firebase.tsx
Comment (mylesdomingo) [inline src/database/firebase.tsx:24]: this should be handled by #15, I would pull from main

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:31]: if there's no tree.name the record shouldn't be included, right? @kennethlien

Comment (mylesdomingo) [reply src/screens/SearchScreen.tsx:31]: @kennethlien if `id` is our primary key shouldn't it be an int?

File: src/screens/TreeScreen.tsx
Comment (mylesdomingo) [reply src/screens/TreeScreen.tsx:22]: to be fixed

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:51]: I would remove the placeholder here

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:49]: i would just pass in the entire tree

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:29]: so using useEffect queries the database everytime you search (eg the dependency on searchQuery). I feel like you don't need to query the database every time, just per render. Is there a reason for this?

File: src/components/SearchCard.tsx
Comment (wangannie) [reply src/components/SearchCard.tsx:7]: i thought it was best practice to pass individual props instead of whole objects (e.g. https://stackoverflow.com/questions/52621169/react-props-should-i-pass-the-object-or-its-properties-does-it-make-much-diffe)

File: src/database/firebase.tsx
Comment (wangannie) [reply src/database/firebase.tsx:24]: A lot of this is because we originally branched off of #15  before it was merged. Git just didn't update the diffs correctly

File: src/screens/SearchScreen.tsx
Comment (wangannie) [reply src/screens/SearchScreen.tsx:29]: oops yeah this was something we meant to revisit but we forgot. we ended up using this logic bc it parallels part of the Recipes onboarding project. i agree we should only query the database once and just filter it on search change. @nikkisuzani  check out  https://www.freecodecamp.org/news/build-a-search-filter-using-react-and-react-hooks/ for how that would work.

File: src/screens/SearchScreen.tsx
Comment (wangannie) [reply src/screens/SearchScreen.tsx:51]: removing this messes with the whole-page styling so we decided to hold off and leave it for now

File: src/screens/SearchScreen.tsx
Comment (kennethlien) [reply src/screens/SearchScreen.tsx:31]: ID is our primary key, and it should be an int. This was previously to allow for existing ID's the NPO already had to not be changed (some keys were "1a"). Let's talk about whether we want to restructure today.

File: src/navigation/TabNavigator.tsx
Comment (kennethlien) [reply src/navigation/TabNavigator.tsx:73]: Yes, @kbrgl is working on those

File: src/components/SearchCard.tsx
Comment (kennethlien) [reply src/components/SearchCard.tsx:7]: That seems like a remnant of JS, but since we're not using all the parts of `Tree` I don't see an issue

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [reply src/screens/SearchScreen.tsx:29]: also since you're first to this can you create a custom hook to fetch the data? eg `useFetchData()`

File: src/screens/SearchScreen.tsx
Comment (wangannie) [reply src/screens/SearchScreen.tsx:19]: Tabling this because @kennethlien will figure out restructuring later!

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:14]: Since `filteredTrees` is directly dependent on `searchQuery`, I believe we don't need a third state variable here.

It might be better to declare a const. eg. 

```
const results = trees.filter(...)
...
{ results.map((tree: Tree)) => (
   <SearchCard key={tree.id} treeName={tree.name} treeID={tree.id} /> 
)}
```

File: src/components/SearchCard.tsx
Comment (mylesdomingo) [reply src/components/SearchCard.tsx:7]: Sure, lets pass individual props then, but declare the type outside of the prop signature and use lookup types:

```
import { Tree } from '...'

type SearchCardProps = {
    name: Tree["name"],
    id: Tree["id"]
}

export default function SearchCard(props: SearchCardProps)
const { name, id } = props;
```

the last part is a nit but I really dislike long / deconstructed signatures.

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:2]: Can you run `yarn prettier --write .` in terminal to fix this?

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [reply src/screens/SearchScreen.tsx:50]: add a width to the search bar / cards to resolve this

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [reply src/screens/SearchScreen.tsx:14]: a little confusing at first `onSearchChange` changes `searchQuery` which changes the `results` on re-render. remove `filteredTrees` altogether

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:30]: nit: @kennethlien I feel like something we're going to encounter is that a lot of our variables will be named `<something>Trees`. This seems redundant and can be reasonably inferred that when we talk about data we mean trees unless otherwise specified, thoughts?

Comment (mylesdomingo) [reply src/screens/SearchScreen.tsx:30]: We should pre-filter out any null or empty-stringed names from our search results, seems rendering them as cards doesn't make sense.

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:31]: See comment above about null-checks.

File: src/screens/SearchScreen.tsx
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:5]: you can combine these into one, eg. 
`import React, { useState, useEffect } from 'react'`

File: src/screens/SearchScreen.tsx
Comment (nikkisuzani) [reply src/screens/SearchScreen.tsx:2]: I tried this and got "error Command failed with exit code 1." and " Usage: prettier [options] [file/dir/glob ...]

By default, output is written to stdout.
Stdin is read if it is piped to Prettier and no files are given." 

how do I fix that?

**Pull request merged**
